tests/scripts/thread-cert/Cert_5_5_01_LeaderReboot.py

```diff
@@ -68,91 +89,131 @@ def test(self):
         self.simulator.go(5)
         self.assertEqual(self.nodes[DUT_LEADER].get_state(), 'leader')
         self.assertEqual(self.nodes[DUT_LEADER].get_addr16(), leader_rloc16)
 
-        # 4 DUT_LEADER: Verify DUT_LEADER sent a multicast Link Request message
-        leader_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        leader_messages_temp = copy.deepcopy(leader_messages)
-
-        msg = leader_messages.next_mle_message(mle.CommandType.LINK_REQUEST)
-        command.check_link_request(
-            msg,
-            tlv_request_address16=command.CheckType.CONTAIN,
-            tlv_request_route64=command.CheckType.CONTAIN,
-        )
-
-        # 5 DUT_ROUTER1: Verify DUT_ROUTER1 replied with Link Accept message
-        router1_messages = self.simulator.get_messages_sent_by(DUT_ROUTER1)
-        router1_messages_temp = copy.deepcopy(router1_messages)
-        msg = router1_messages.next_mle_message(mle.CommandType.LINK_ACCEPT)
-        if msg is not None:
-            command.check_link_accept(
-                msg,
-                self.nodes[DUT_LEADER],
-                address16=command.CheckType.CONTAIN,
-                leader_data=command.CheckType.CONTAIN,
-                route64=command.CheckType.CONTAIN,
-            )
-        else:
-            msg = router1_messages_temp.next_mle_message(mle.CommandType.LINK_ACCEPT_AND_REQUEST)
-            self.assertTrue(msg is not None)
-            command.check_link_accept(
-                msg,
-                self.nodes[DUT_LEADER],
-                address16=command.CheckType.CONTAIN,
-                leader_data=command.CheckType.CONTAIN,
-                route64=command.CheckType.CONTAIN,
-                challenge=command.CheckType.CONTAIN,
-            )
-
-        # 6 DUT_LEADER: Verify DUT_LEADER didn't send a Parent Request message
-        msg = leader_messages_temp.next_mle_message(mle.CommandType.PARENT_REQUEST, False)
-        self.assertTrue(msg is None)
-
-        # 7 ALL: Verify connectivity by sending an ICMPv6 Echo Request from
-        # DUT_LEADER to DUT_ROUTER1 link local address
         router1_link_local_address = self.nodes[DUT_ROUTER1].get_ip6_address(config.ADDRESS_TYPE.LINK_LOCAL)
         self.assertTrue(self.nodes[DUT_LEADER].ping(router1_link_local_address))
 
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        ROUTER = pv.vars['ROUTER']
+
+        # Step 1: Verify topology is formed correctly.
+        pv.verify_attached('ROUTER', 'LEADER')
+
+        # Step 2: The DUT MUST send properly formatted MLE Advertisements with
+        #         an IP Hop Limit of 255 to the Link-Local All Nodes multicast
+        #         address (FF02::1).
+        #          The following TLVs MUST be present in the MLE Advertisements:
+        #              - Leader Data TLV
+        #              - Route64 TLV
+        #              - Source Address TLV
```

Comment:
Should we validate these TLVs are present?

Reply:
yes, validated in filter_mle_advertisement https://github.com/openthread/openthread/blob/d218dfcb354309600c379e307ea4742236f0bacc/tests/scripts/thread-cert/pktverify/packet_filter.py#L354